### PR TITLE
DO_NOT_MERGE: build: improve cmake

### PR DESCRIPTION
**Pull request by gocarlos**

### CODE REVIEW

**Comment by gocarlos** [comment]
https://github.com/jimsch/cn-cbor/pull/9 should be merged first

File: CMakeLists.txt [hunk 1]
```diff
@@ -46,10 +47,14 @@ if (NOT CMAKE_BUILD_TYPE)
       set ( CMAKE_BUILD_TYPE MinSizeRel )
       set ( coveralls OFF )
       set ( coveralls_send OFF )
-   else ()
-      set ( CMAKE_BUILD_TYPE Debug)
+   else()
+      message(STATUS "CMAKE_BUILD_TYPE empty setting to DEBUG")
+      set(CMAKE_BUILD_TYPE "Debug")
+      if(coveralls)
+         link_libraries(-lgcov)
```
Comment (jimsch) [inline CMakeLists.txt]: If coveralls gets turned off at line 82, this will be a problem?

File: conanfile.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def source(self):
+        self.run(
+            "git clone -b build--improve-cmake git@github.com:gocarlos/COSE-C.git")
```
Comment (jimsch) [inline conanfile.py]: Does this need changing now that I have merged in over there?

Comment (gocarlos) [reply conanfile.py]: I'll change this

**Comment by gocarlos** [comment]
I'll rebase this, too may commits, I'll let you know when its ready

File: test/encrypt.c [hunk 3]
```diff
@@ -425,8 +425,7 @@ int _ValidateEncrypt(const cn_cbor * pControl, const byte * pbEncoded, size_t cb
 		else if ((pFail != NULL) && (pFail->type != CN_CBOR_TRUE)) fFail = true;
 
                 size_t cb;
-                byte * pb;
-                pb = COSE_Encrypt_GetContent(hEnc, &cb, NULL);
+                COSE_Encrypt_GetContent(hEnc, &cb, NULL);
```
Comment (gocarlos) [inline test/encrypt.c:428]: unused var

File: test/ReadMe.txt [hunk 4]
```diff
@@ -1,40 +0,0 @@
...
-    These files are used to build a precompiled header (PCH) file
-    named test.pch and a precompiled types file named StdAfx.obj.
-
-/////////////////////////////////////////////////////////////////////////////
-Other notes:
-
-AppWizard uses "TODO:" comments to indicate parts of the source code you
-should add to or customize.
-
-/////////////////////////////////////////////////////////////////////////////
```
Comment (gocarlos) [inline test/ReadMe.txt:40]: old stuff I guess

**Pull request closed without merging**